### PR TITLE
changes based on feedback

### PR DESCRIPTION
**Pull request by smcgarrity**
* Update custard command to not take parameters but identify the current and target versions based on current application state
* Remove resume point
* Remove priority numbers
* Reduce use of `RunMigrationsEntity`, particularly outside of `RunMigrationsUseCase`
* Improve and expand tests

### CODE REVIEW

**Comment by smcgarrity** [comment]
acuthbert, one area I was not sure if went far enough with was reducing the use of the Entity within the RunMIgrationUseCase.

File: src/MigrationsModule.php [hunk 1]
```diff
@@ -19,21 +19,36 @@
...
-use Rhubarb\Modules\Migrations\Commands\MigrateCommand;
+use Rhubarb\Modules\Migrations\Commands\GetMigrationStateCommand;
+use Rhubarb\Modules\Migrations\Commands\RunMigrationsCommand;
 use Rhubarb\Modules\Migrations\Commands\RunSingleMigrationScriptCommand;
 
 class MigrationsModule extends Module
 {
+    protected $providerClass;
+
+    public function __construct(string $providerClass)
```
Comment (acuthbert) [inline src/MigrationsModule.php]: Small thing - but maybe this could be $migrationStateProviderClass?

File: src/UseCases/RunMigrationsEntity.php [hunk 2]
```diff
@@ -3,26 +3,16 @@
...
 namespace Rhubarb\Modules\Migrations\UseCases;
 
-
-use Rhubarb\Modules\Migrations\MigrationsManager;
-use Rhubarb\Modules\Migrations\MigrationsSettings;
 use Rhubarb\Modules\Migrations\MigrationsStateProvider;
 use Rhubarb\Modules\Migrations\Interfaces\MigrationScriptInterface;
 
-class MigrationEntity
+class RunMigrationsEntity
```
Comment (acuthbert) [inline src/UseCases/RunMigrationsEntity.php]: Hi Sean, I'd be sorely tempted to scrap this class now. It only has 3 values which would be simpler to pass as arguments. Also see comment below about the use of the provider from within.

File: src/UseCases/RunMigrationsEntity.php [hunk 3]
```diff
@@ -31,7 +21,7 @@ class MigrationEntity
 
     public function __construct()
     {
-        // Default Values
         $this->startVersion = MigrationsStateProvider::getProvider()->getLocalVersion();
+        $this->endVersion = MigrationsStateProvider::getApplicationVersion();
```
Comment (acuthbert) [inline src/UseCases/RunMigrationsEntity.php]: Yeah I think the issue here is that an entity is reaching beyond itself to 'find' it's values. An entity is always dumb - and should be configured externally. The custard command or use case could do this - but given my comment above, I think I'd prefer to scrap this class altogether.

**Review by acuthbert — COMMENTED** [review]
Almost there Sean, just need to get documentation in, and look at the few comments here.

**Comment by acuthbert** [comment]
Great thanks Sean!

**Pull request merged**